components/Tokens/TokenList.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import TokenListItem from '@/components/Tokens/TokenListItem';
+import { useTokenList } from '@/hooks/useTokenList';
+import Link from 'next/link';
+
+const TokenList: React.FC = () => {
+    const onSuccess = () => {
+        console.log('Perform side effect after data fetching.');
+    }
+    const onError = () => {
+        console.log('Perform side effect after encountering an error.')
+    }
+
+    const {isLoading, data, isError, error, isFetching, refetch} = useTokenList(onSuccess, onError);
+
+    console.log(isLoading, isFetching);
+
+    if(isLoading || isFetching) {
+        return <div>Loading...</div>;
+    }
+
+    if(isError) {
+        return <div>{error?.message}</div>
+    }
+
```

Comment:
This is indeed the power of react query hooks, it makes handling data states so much easier and cleaner to read. Good code